eslint.config.mjs

```diff
@@ -188,88 +169,103 @@ export default tseslint.config([
+              ? { type: 'natural' }
+              : { partitionByComment: true, type: 'natural' },
+          ],
+        ]),
+      ),
       'prefer-arrow-callback': 'off',
       'prefer-const': ['error', { destructuring: 'all' }],
       'prefer-destructuring': 'off', // TODO: Consider enabling this
       'prefer-named-capture-group': 'off',
       'prefer-object-has-own': 'off', // TODO: Consider enabling this
       'prefer-template': 'off',
-      'new-cap': 'off',
-      'no-await-in-loop': 'off',
-      'no-console': ['error', { allow: ['warn', 'error', 'table', 'trace'] }],
-      'no-continue': 'off',
-      'no-duplicate-imports': 'error',
-      'no-else-return': 'off',
-      'no-eq-null': 'off', // TODO: Consider enabling this
-      'no-inline-comments': 'off',
-      'no-loop-func': 'off',
-      'no-nested-ternary': 'off',
-      'no-magic-numbers': 'off',
-      'no-param-reassign': 'off',
-      'no-plusplus': 'off', // TODO: Consider enabling this
-      'no-ternary': 'off',
-      'no-underscore-dangle': 'off',
-      'no-use-before-define': 'off',
-      'no-warning-comments': 'off',
-      'no-undef': 'off',
-      'no-undef-init': 'off', // TODO: Consider enabling this
-      'no-undefined': 'off',
-      'no-unused-vars': 'off',
-      'max-classes-per-file': 'off',
-      'max-depth': 'off',
-      'max-lines': 'off',
-      'max-lines-per-function': 'off',
-      'max-params': 'off',
-      'max-statements': 'off',
       radix: ['error', 'as-needed'],
       'require-atomic-updates': 'off',
       'require-await': 'off', // TODO: Consider enabling this
       'require-unicode-regexp': 'off',
+
       'sort-vars': 'off',
 
       // Enforce alphabetical order of import specifiers within each import group.
       // The import-x/order rule handles the overall sorting of the import groups.
-      'sort-imports': [
-        'error',
-        {
-          ignoreDeclarationSort: true,
-          ignoreMemberSort: false,
-          memberSyntaxSortOrder: ['none', 'all', 'multiple', 'single'],
-        },
-      ],
-      'sort-keys': 'off',
-
       'import-x/order': [
         'error',
         {
-          'newlines-between': 'always',
-
           alphabetize: {
             order: 'asc',
           },
 
+          'newlines-between': 'always',
+
           pathGroups: [
             {
-              pattern: '@prairielearn/**',
               group: 'external',
+              pattern: '@prairielearn/**',
               position: 'after',
             },
           ],
 
           pathGroupsExcludedImportTypes: ['builtin'],
         },
       ],
-
       'no-floating-promise/no-floating-promise': 'error',
 
+      'sort-imports': [
+        'error',
+        {
+          ignoreDeclarationSort: true,
+          ignoreMemberSort: false,
+          memberSyntaxSortOrder: ['none', 'all', 'multiple', 'single'],
+        },
+      ],
+
+      'sort-keys': 'off',
```

Comment:
Nit: why pull in the "perfectionist" plugin? I haven't looked at this in depth, but `sort-keys` would seem to do what we want.